pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -52,20 +61,15 @@
     @VisibleForTesting
     final Map<String, Map<String, Map<String, FunctionMetaData>>> functionMetaDataMap = new ConcurrentHashMap<>();
 
-    // A map in which the key is the service request id and value is the service request
-    private final Map<String, ServiceRequestInfo> pendingServiceRequests = new ConcurrentHashMap<>();
-
     private final ServiceRequestManager serviceRequestManager;
     private final SchedulerManager schedulerManager;
     private final WorkerConfig workerConfig;
     private final PulsarClient pulsarClient;
     private final ErrorNotifier errorNotifier;
 
     private FunctionMetaDataTopicTailer functionMetaDataTopicTailer;
-
-    @Setter
-    @Getter
-    boolean isInitializePhase = false;
+    private Producer exclusiveLeaderProducer;
+    private MessageId lastMessageSeen = MessageId.earliest;
```

Comment:
It seems more natural that the FunctionMetaDataTopicTailer keeps track of this

Reply:
Actually no. Manager should keep track of it since in leader mode there is no tailer